src/imperative/cached_op.cc

```diff
@@ -855,10 +855,15 @@ OpStatePtr CachedOp::Forward(
   int prev_bulk_size = Engine::Get()->set_bulk_size(config_.forward_bulk_size);
 
   OpStatePtr op_state;
-  if (config_.static_alloc) {
-    op_state = StaticForward(default_ctx, inputs, outputs);
-  } else {
-    op_state = DynamicForward(default_ctx, inputs, outputs);
+  try {
```

Comment:
This change looks good to me. One nit suggestion would be: is it cleaner by using a RAII class object to manage the bulk_size value instead of a local variable? Doing that we may also get rid of the try block. Not something necessary, but I would love to hear some comments on this.

Reply:
Thanks for the suggestion. It can be done but try-catch also gives us more flexibility to do more things (like your comment above which we need to do more cleanup upon exception). Prefer to keep try-catch here.